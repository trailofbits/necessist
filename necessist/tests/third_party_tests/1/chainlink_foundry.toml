url = "https://github.com/smartcontractkit/chainlink"
rev = "v2.3.0"
# smoelius: The next `pnpm up` is needed to get the packages to build on macOS.
init = "cd contracts && pnpm up ethereumjs-abi"
subdir = "contracts"
framework = "foundry"
full = true

[config]
ignored_functions = ["vm.prank", "vm.startPrank", "vm.stopPrank"]
