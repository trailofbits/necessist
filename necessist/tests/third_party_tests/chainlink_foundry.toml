url = "https://github.com/smartcontractkit/chainlink"
subdir = "contracts"
framework = "foundry"
full = true
