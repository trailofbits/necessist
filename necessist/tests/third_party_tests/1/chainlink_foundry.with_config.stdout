18 candidates in 2 test files
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol: dry running
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol: mutilating
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:42:5-42:40: `s_coordinator.createSubscription();` failed
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:42:18-42:39: `.createSubscription()` failed
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:70:5-70:40: `s_coordinator.createSubscription();` failed
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:70:18-70:39: `.createSubscription()` failed
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:80:42-80:59: `.balanceOf(OWNER)` nonbuildable
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:87:5-87:82: `(balance, requestCount, owner, consumers) = s_coordinator.getSubscription(1);` failed
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:95:41-95:58: `.balanceOf(OWNER)` nonbuildable
$DIR/contracts/test/v0.8/foundry/dev/special/NoCancelVRFCoordinatorV2.t.sol:102:47-106:6: `.calculatePaymentAmountTest(
      gasAfterPaymentCalculation,
      fulfillmentFlatFeeLinkPPM,
      weiPerUnitGas
    )` nonbuildable
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol: dry running
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol: mutilating
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:26:22-26:39: `.typeAndVersion()` nonbuildable
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:36:36-36:53: `.encode("newKey")` nonbuildable
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:39:49-39:67: `.getDONPublicKey()` nonbuildable
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:45:5-45:44: `s_oracle.setDONPublicKey(donPublicKey);` failed
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:45:13-45:43: `.setDONPublicKey(donPublicKey)` failed
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:46:44-46:62: `.getDONPublicKey()` nonbuildable
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:77:47-77:61: `.getRegistry()` nonbuildable
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:81:5-81:43: `s_oracle.setRegistry(registryAddress);` failed
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:81:13-81:42: `.setRegistry(registryAddress)` failed
$DIR/contracts/test/v0.8/foundry/functions/FunctionsOracle.t.sol:82:42-82:56: `.getRegistry()` nonbuildable
